mlrun/model.py

```diff
@@ -1260,7 +1261,7 @@ def new_task(
         run = deepcopy(base)
     else:
         run = RunTemplate()
-    run.metadata.name = name or run.metadata.name
+    run.metadata.name = normalize_name(name or run.metadata.name or "")
```

Comment:
why "" ?

Reply:
if both are None when `normalize_name ` will fail.. I can also do 
`normalize_name(name or run.metadata.name) if name or run.metadata.name else None`

Reply:
but why not failing early and raising an error if no name is provided?

Reply:
ah, I see what u are saying that in the `function.run()` we will have name for the function and therefore there is no need to fail now?